src/build_workflow/build_recorder.py

```diff
@@ -59,6 +59,7 @@ def __init__(self, target: BuildTarget, build_manifest: BuildManifest = None) ->
 
             if build_manifest:
                 self.data = build_manifest.__to_dict__()
+                self.data["build"]["id"] = target.build_id
```

Comment:
Is this missed in the last PR?
What is it used for here?

Reply:
The new generated build manifest would keep the old manifest build id even after incremental building some of components. This change will make sure the new build manifest will have a new build ID; in CI, this will be the new job number.

However, due to the logic of incremental build, if nothing changes, build ID won't change.

Reply:
Good catch! Maybe add a test for this @zelinh ? This could have serious wrong results when running integration tests because all we pass is the buidl manifest.

Reply:
Test for this case is added.

Reply:
The issue we discussed above is resolved in the latest commit. Please help review it when you have time. Thanks! 
@gaiksaya @peterzhuamazon